payments-ui-core/src/main/java/com/stripe/android/ui/core/forms/TransformSpecToElements.kt

```diff
@@ -35,14 +35,18 @@ import com.stripe.android.ui.core.forms.resources.ResourceRepository
  * Transform a [LayoutSpec] data object into an Element, which
  * has a controller and identifier.  With only a single field in a section the section
  * controller will be a pass through the field controller.
+ *
+ * @param viewOnlyFields A set of identifiers for the fields that should be view-only, non-editable.
+ * Currently only [IdentifierSpec.CardNumber] is supported and any other identifier is ignored.
  */
 class TransformSpecToElements(
     private val resourceRepository: ResourceRepository,
     private val initialValues: Map<IdentifierSpec, String?>,
     private val amount: Amount?,
     private val saveForFutureUseInitialValue: Boolean,
     private val merchantName: String,
-    private val context: Context
+    private val context: Context,
+    private val viewOnlyFields: Set<IdentifierSpec> = emptySet()
```

Comment:
Is this actually implemented globallly.  Like if I set Name to be viewOnly will it set the enabled field in the TextFieldController!!  That would be awesome.

If not, I would make it more clear what exactly it is controlling.

Reply:
The idea is to make this generic so that what you mentioned happens. But I'm not going through all elements now implementing their view-only versions, so the comment explains that only `IdentifierSpec.CardNumber` is currently supported. I think this is better than for example adding a boolean `showCardNumberAsReadOnly`. Any suggestions on how this could be more clear?

Reply:
I did not estimate that as it's out of the scope of what I need. The effort for the card number was considerable, so I guess it would be at least half a week of work.